src/Explorer/Controls/Settings/SettingsUtils.tsx

```diff
@@ -0,0 +1,331 @@
+import * as React from "react";
+import * as AutoPilotUtils from "../../../Utils/AutoPilotUtils";
+import * as SharedConstants from "../../../Shared/Constants";
+import * as CommonConstants from "../../../Common/Constants";
+import * as DataModels from "../../../Contracts/DataModels";
+import { AutopilotTier } from "../../../Contracts/DataModels";
+import {
+  computeAutoscaleUsagePriceHourly,
+  getPriceCurrency,
+  getCurrencySign,
+  getAutoscalePricePerRu,
+  getMultimasterMultiplier,
+  computeRUUsagePriceHourly,
+  getPricePerRu,
+  calculateEstimateNumber
+} from "../../../Utils/PricingUtils";
+
+/*
+Class used to track baseline and current values of a state object. If both the values differ,
```

Comment:
This seems like a generally useful utility that we could reuse elsewhere. Can we move it to its own file?
A class would be better suited, so we can call `isDirty()` on it (and possibly pass the `updateCurrent`) parameter when updating the value. I would make it generic so we can pass our own type, rather than be limited to `StatefulValuesType`. We can pass `StatefulValuesType` in this case.